src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -4531,44 +4541,48 @@ class MavlinkStreamDistanceSensor : public MavlinkStream
 
 	bool send(const hrt_abstime t) override
 	{
-		distance_sensor_s dist_sensor;
+		auto updated = false;
 
-		if (_distance_sensor_sub.update(&dist_sensor)) {
-			mavlink_distance_sensor_t msg{};
+		for (int i = 0; i < ORB_MULTI_MAX_INSTANCES; i++) {
+			distance_sensor_s dist_sensor;
 
-			msg.time_boot_ms = dist_sensor.timestamp / 1000; /* us to ms */
+			if (_distance_sensor_sub[i].update(&dist_sensor)) {
+				mavlink_distance_sensor_t msg{};
 
-			switch (dist_sensor.type) {
-			case MAV_DISTANCE_SENSOR_ULTRASOUND:
-				msg.type = MAV_DISTANCE_SENSOR_ULTRASOUND;
-				break;
+				msg.time_boot_ms = dist_sensor.timestamp / 1000; /* us to ms */
 
-			case MAV_DISTANCE_SENSOR_LASER:
-				msg.type = MAV_DISTANCE_SENSOR_LASER;
-				break;
+				switch (dist_sensor.type) {
+				case MAV_DISTANCE_SENSOR_ULTRASOUND:
+					msg.type = MAV_DISTANCE_SENSOR_ULTRASOUND;
+					break;
 
-			case MAV_DISTANCE_SENSOR_INFRARED:
-				msg.type = MAV_DISTANCE_SENSOR_INFRARED;
-				break;
+				case MAV_DISTANCE_SENSOR_LASER:
+					msg.type = MAV_DISTANCE_SENSOR_LASER;
+					break;
 
-			default:
-				msg.type = MAV_DISTANCE_SENSOR_LASER;
-				break;
-			}
+				case MAV_DISTANCE_SENSOR_INFRARED:
+					msg.type = MAV_DISTANCE_SENSOR_INFRARED;
+					break;
 
-			msg.current_distance = dist_sensor.current_distance * 1e2f; // m to cm
-			msg.id               = dist_sensor.id;
-			msg.max_distance     = dist_sensor.max_distance * 1e2f;     // m to cm
-			msg.min_distance     = dist_sensor.min_distance * 1e2f;     // m to cm
-			msg.orientation      = dist_sensor.orientation;
-			msg.covariance       = dist_sensor.variance * 1e4f;         // m^2 to cm^2
+				default:
+					msg.type = MAV_DISTANCE_SENSOR_LASER;
+					break;
+				}
 
-			mavlink_msg_distance_sensor_send_struct(_mavlink->get_channel(), &msg);
+				msg.current_distance = dist_sensor.current_distance * 1e2f; // m to cm
+				msg.id               = dist_sensor.id;
```

Comment:
For the sake of mavlink I'm wondering if this id should be the instance?

Reply:
I think that's the only thing that makes sense because it's a uint8.

Reply:
Actually, in that case you should revert the for loop change above to keep the index, then set `msg.id = i;` here.

Reply:
That might make sense. The ID for LL40LS as an example, is currently from what I can tell computed somehow based on a combination of its configuration, such as bus number and sensor direction. Using the instance number seems reasonable. I'll make the change.

Reply:
The actual device id (uint32_t) we use for accel/gyro/mag calibration comes from here. https://github.com/PX4/Firmware/blob/df38e3798c25541d1462d8011fbf0d58904ce4f7/src/lib/drivers/device/Device.hpp#L166-L171

For various reasons like per sensor configuration (orientation, position) I'd like to get that in place for all the distance sensors, but it's still not going to help for mavlink because the id field is only 1 byte.